### ISSUE TITLE
Implement BOM

### ISSUE THREAD

**Issue by sophiewigmore** [paketo-buildpacks/npm-install#261]
As a result of #218, we have a viable way to implement the BOM on the NPM Install buildpack so that we can collect metadata on all language-specific modules installed by this buildpack.

This issue is to implement this work as it will be outlined in the RFC associated with https://github.com/paketo-buildpacks/nodejs/issues/425.

## Acceptance
The outcome of this story should be that a BOM is available from this buildpack that includes metadata about all of the installed modules. The format should conform to https://github.com/paketo-buildpacks/rfcs/pull/57

## Blocking
This story is blocked on the creation and acceptance of the RFC in https://github.com/paketo-buildpacks/nodejs/issues/425.

**Comment by sophiewigmore** [comment]
Closing in favour of https://github.com/paketo-buildpacks/node-module-bom/issues/1 due to changes to https://github.com/paketo-buildpacks/rfcs/pull/98 implementation details.
